src/metabase/lib/legacy_metric.cljc

```diff
@@ -112,15 +112,19 @@
                                        (keep-indexed (fn [index aggregation-clause]
                                                        (when (lib.util/clause-of-type? aggregation-clause :metric)
                                                          [(get aggregation-clause 2) index])))
-                                       (lib.aggregation/aggregations query stage-number))]
-         (cond
-           (empty? metrics)             nil
-           (empty? metric-aggregations) (vec metrics)
-           :else                        (mapv (fn [metric-metadata]
-                                                (let [aggregation-pos (-> metric-metadata :id metric-aggregations)]
-                                                  (cond-> metric-metadata
-                                                    aggregation-pos (assoc :aggregation-position aggregation-pos))))
-                                              metrics)))))))
+                                       (lib.aggregation/aggregations query stage-number))
+             results (cond
+                       (empty? metrics)             nil
+                       (empty? metric-aggregations) (vec metrics)
+                       :else                        (mapv (fn [metric-metadata]
+                                                            (let [aggregation-pos (-> metric-metadata :id metric-aggregations)]
+                                                              (cond-> metric-metadata
+                                                                aggregation-pos (assoc :aggregation-position aggregation-pos))))
+                                                          metrics))
+             sorted (sort-by (some-fn :display-name :name) results)]
+         (if (empty? sorted)
+           nil
+           sorted))))))
```

Comment:
If you can, the following form would be more concise:
```suggestion
         (not-empty sorted))))))
```

Reply:
Otherwise I'd do the following to ensure vector is returned:
```
(when (seq sorted)
  (vec sorted))
```

Reply:
Or rather wrap call to `sort-by` in `(vec ...). Then, also the `not-empty` version would do.